app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,49 @@
+
+class Api::V1::BaseController < ApplicationController
+  before_action :retrieve_source_model_by_id, only: [:edit, :update, :destroy, :show]
+
+  def index
+    @source_models = source_klass.all
```

Comment:
Yes, but the CRUD operations should also be implemented here, right? I'll remove the actions I don't use.